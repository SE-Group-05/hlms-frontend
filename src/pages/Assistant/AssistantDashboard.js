import React from "react";
import { Box, Center, Heading, Divider, Wrap, WrapItem } from "@chakra-ui/react";
import PreviewCard from "../../components/Assistant/MenuPreviewCard";


function AssistantDashboard() {

    return (
        <>
            <Center w="100%" h="100%" p={4} pb='100px'>
                <Box w="90%" h="auto" borderWidth="1px" borderRadius="lg" p='30px'>
                    <Heading as="h2" size="2xl" ml='20px' mb='20px'>Assitant Dashboard</Heading>
                    <Heading as="h4" size="md" ml='20px' mb='30px' >Sunrise - Hotel Assistant UI</Heading>
                    <Divider />
                    <Wrap spacing='50px' justify="center" mt='20px' mb='10px'>
                        <WrapItem>
                            <PreviewCard
                                imageUrl="https://blogmedia.evbstatic.com/wp-content/uploads/wpmulti/sites/8/2017/04/21143242/tourists.jpg"
                                imageAlt="Tourist Couple Picture"
                                title="Tourist Manager "
                                linkTo="/app/tourists"
                                description=" Add Tousrists, Delete Tourists, Update Tourists Information and View Activites"
                            />
                        </WrapItem>
                        <WrapItem>

                            <PreviewCard
                                imageUrl="https://trendipia.com/wp-content/uploads/2017/05/phuket-thailand-beach-800x416.jpeg"
                                imageAlt="Location Picture"
                                title="View Visiting Place"
                                linkTo="/app/assistantvp"
                                description=" Search Visiting Places"
                            />


                        </WrapItem>
                        <WrapItem>

                            <PreviewCard
                                imageUrl="https://web-static.wrike.com/cdn-cgi/image/width=880,format=auto,q=80/blog/content/uploads/2020/01/Five-Features-of-a-Good-Monthly-Employee-Work-Schedule-Template.jpg?av=unknown"
                                imageAlt="Tourist Couple Picture"
                                title="Schedule Manager"
                                linkTo="/app/schedules"
                                description=" View Schedules"
                            />


                        </WrapItem>


                    </Wrap>


                </Box>
            </Center>

        </>
    );
}

export default AssistantDashboard;