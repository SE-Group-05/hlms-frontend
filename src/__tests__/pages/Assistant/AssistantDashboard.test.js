// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, cleanup, waitFor } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import AssistantDashboard from "../../../pages/Assistant/AssistantDashboard";
import { MemoryRouter } from "react-router-dom";

afterEach(cleanup);


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter><AssistantDashboard /></MemoryRouter>,
    div
  );
});



it("matches snapshot", () => {
  const tree = renderer.create(
    <MemoryRouter><AssistantDashboard /></MemoryRouter>
  );

  expect(tree).toMatchSnapshot();
});
