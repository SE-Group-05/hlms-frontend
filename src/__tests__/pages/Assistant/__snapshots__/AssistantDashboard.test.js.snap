// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-wcxhkq"
>
  <div
    className="css-102niab"
  >
    <h2
      className="chakra-heading css-1h40200"
    >
      Assitant Dashboard
    </h2>
    <h4
      className="chakra-heading css-x05dfc"
    >
      Sunrise - Hotel Assistant UI
    </h4>
    <hr
      aria-orientation="horizontal"
      className="chakra-divider css-1upb9tn"
      role="separator"
    />
    <div
      className="chakra-wrap css-1nvos36"
    >
      <ul
        className="chakra-wrap__list css-rhkfrd"
      >
        <li
          className="chakra-wrap__listitem css-y9uxwf"
        >
          <a
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "to": [Function],
                },
                "render": [Function],
              }
            }
            href="/app/tourists"
            onClick={[Function]}
          >
            <div
              className="css-17xbt85"
              data-testid="MenuPreviewCard"
            >
              <img
                alt="Tourist Couple Picture"
                className="chakra-image__placeholder css-1rdn362"
              />
              <div
                className="css-mfzh6r"
              >
                <div
                  className="css-0"
                >
                  <div
                    className="css-76gc8x"
                    data-testid="title"
                  >
                    Tourist Manager 
                  </div>
                  <div
                    className="chakra-collapse"
                    style={
                      Object {
                        "display": "none",
                        "height": "0px",
                        "opacity": 0,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <div
                      className="css-j5j99d"
                    >
                       Add Tousrists, Delete Tourists, Update Tourists Information and View Activites
                    </div>
                  </div>
                  <div
                    className="css-k008qs"
                  >
                    <div
                      className="css-0"
                    />
                    <div
                      className="css-17xejub"
                    />
                    <div
                      className="css-0"
                    >
                      <button
                        className="chakra-button css-143x6ju"
                        onClick={[Function]}
                        type="button"
                      >
                        Show 
                        More
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </li>
        <li
          className="chakra-wrap__listitem css-y9uxwf"
        >
          <a
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "to": [Function],
                },
                "render": [Function],
              }
            }
            href="/app/assistantvp"
            onClick={[Function]}
          >
            <div
              className="css-17xbt85"
              data-testid="MenuPreviewCard"
            >
              <img
                alt="Location Picture"
                className="chakra-image__placeholder css-1rdn362"
              />
              <div
                className="css-mfzh6r"
              >
                <div
                  className="css-0"
                >
                  <div
                    className="css-76gc8x"
                    data-testid="title"
                  >
                    View Visiting Place
                  </div>
                  <div
                    className="chakra-collapse"
                    style={
                      Object {
                        "display": "none",
                        "height": "0px",
                        "opacity": 0,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <div
                      className="css-j5j99d"
                    >
                       Search Visiting Places
                    </div>
                  </div>
                  <div
                    className="css-k008qs"
                  >
                    <div
                      className="css-0"
                    />
                    <div
                      className="css-17xejub"
                    />
                    <div
                      className="css-0"
                    >
                      <button
                        className="chakra-button css-143x6ju"
                        onClick={[Function]}
                        type="button"
                      >
                        Show 
                        More
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </li>
        <li
          className="chakra-wrap__listitem css-y9uxwf"
        >
          <a
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "to": [Function],
                },
                "render": [Function],
              }
            }
            href="/app/schedules"
            onClick={[Function]}
          >
            <div
              className="css-17xbt85"
              data-testid="MenuPreviewCard"
            >
              <img
                alt="Tourist Couple Picture"
                className="chakra-image__placeholder css-1rdn362"
              />
              <div
                className="css-mfzh6r"
              >
                <div
                  className="css-0"
                >
                  <div
                    className="css-76gc8x"
                    data-testid="title"
                  >
                    Schedule Manager
                  </div>
                  <div
                    className="chakra-collapse"
                    style={
                      Object {
                        "display": "none",
                        "height": "0px",
                        "opacity": 0,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <div
                      className="css-j5j99d"
                    >
                       View Schedules
                    </div>
                  </div>
                  <div
                    className="css-k008qs"
                  >
                    <div
                      className="css-0"
                    />
                    <div
                      className="css-17xejub"
                    />
                    <div
                      className="css-0"
                    >
                      <button
                        className="chakra-button css-143x6ju"
                        onClick={[Function]}
                        type="button"
                      >
                        Show 
                        More
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
